File: Java_MVVM/app/src/main/java/eu/gotoinc/requesinjava_mvvm/adapter/UserDataSourceFactory.java
package eu.gotoinc.requesinjava_mvvm.adapter;

import androidx.paging.DataSource;
import eu.gotoinc.requesinjava_mvvm.util.architecture.SingleLiveEvent;
import eu.gotoinc.requesinjava_mvvm.repository.ApiContract;
import eu.gotoinc.requesinjava_mvvm.repository.data_models.responses.UserResponse;

public class UserDataSourceFactory extends DataSource.Factory<Integer, UserResponse> {

	private final ApiContract api;
	private final SingleLiveEvent<Integer> liveErrorEvent;

	public UserDataSourceFactory(ApiContract api, SingleLiveEvent<Integer> liveErrorEvent) {
		this.api = api;
		this.liveErrorEvent = liveErrorEvent;
	}

	@Override
	public UserPagedDataSource create() {
		return new UserPagedDataSource(api, liveErrorEvent);
	}
}

File: Kotlin_MVVM/app/src/main/java/eu/gotoinc/kotlinmvvm/adapter/UserDataSourceFactory.kt
package eu.gotoinc.kotlinmvvm.adapter

import androidx.paging.DataSource
import eu.gotoinc.kotlinmvvm.repository.ApiContract
import eu.gotoinc.kotlinmvvm.repository.datamodel.response.UserResponse
import eu.gotoinc.kotlinmvvm.util.architecture.SingleLiveEvent

class UserDataSourceFactory(private val api: ApiContract, private val liveErrorEvent: SingleLiveEvent<Int>) :
    DataSource.Factory<Int, UserResponse>() {

    override fun create(): UserPagedDataSource = UserPagedDataSource(api, liveErrorEvent)

}